lib/Formatter.php

```diff
@@ -31,6 +31,13 @@ public static function RemoveDiacritics(string $text): string{
 		}
 	}
 
+	/**
+	 * Remove diacritics and non-alphanumeric characters.
+	 */
+	public static function RemoveDiacriticsAndNonalphanumerics(string $text): string{
+		return trim(preg_replace('|[^a-zA-Z0-9 ]|ius', ' ', Formatter::RemoveDiacritics($text)));
```

Comment:
Are you sure we want to replaced with `' '` and not `''`? Introducing spaces may unexpectedly change how matches occur.